packages/features/calendars/weeklyview/components/event/Event.tsx

```diff
@@ -19,27 +49,32 @@ export function Event({
   onEventClick,
 }: EventProps) {
   const selected = currentlySelectedEventId === event.id;
+  const { options } = event;
+
+  const borderColor = options?.borderColor ? "custom" : options?.status;
+
+  const styles = options?.borderColor
+    ? {
+        borderColor: options?.borderColor,
+      }
+    : {};
 
   const Component = onEventClick ? "button" : "div";
 
   return (
     <Component
       onClick={() => onEventClick?.(event)} // Note this is not the button event. It is the calendar event.
-      className={classNames(
-        "group  flex h-full w-full flex-col overflow-y-auto rounded-[4px]  px-[6px] py-1 text-xs font-semibold  leading-5 ",
-        event.status === "ACCEPTED" &&
-          !selected &&
-          "bg-subtle hover:bg-emphasis text-emphasis border-[1px] border-gray-900",
-        event.status === "PENDING" &&
-          !selected &&
-          "bg-default text-emphasis border-[1px] border-dashed border-gray-900",
-        selected && "bg-inverted text-inverted border-[1px] border-transparent",
-        disabled ? "hover:cursor-default" : "hover:cursor-pointer"
-      )}>
+      className={eventClasses({
+        status: options?.status,
+        disabled,
+        selected,
+        borderColor,
+      })}
+      style={styles}>
       <div className="w-full overflow-hidden overflow-ellipsis whitespace-nowrap text-left leading-4">
         {event.title}
       </div>
-      {eventDuration >= 30 && (
+      {eventDuration > 30 && (
```

Comment:
30 min time slots were being cut off from overflow. Removing the displayed time if the event is 30mins and only show > than